core/common/src/test/java/alluxio/AlluxioTestDirectory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import com.google.common.base.Throwables;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.AgeFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.UUID;
+
+/**
+ * Class for managing creation and cleanup of an Alluxio test directory.
+ *
+ * The Alluxio test directory is created as a subdirectory of the system temp directory, and on
+ * class initialization it deletes files and directories older than the maximum age.
+ */
+public final class AlluxioTestDirectory {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static final int MAX_FILE_AGE_HOURS = 2;
```

Comment:
The runs are under the same /tmp/alluxio-tests/ directory. For example, the pull request builder might create /tmp/alluxio-tests/folderA at 1pm, then while that's running the master build starts at 1:50pm. If the threshold was under 50 minutes the master build would delete /tmp/alluxio-tests/folderA out from under the pull request builder.

Basically, we just want a safe value here such that we don't delete any directories that are actually being used. We could probably get away with lowering this to ~30 minutes, but I chose 2 hours to be completely safe. I'll change it to 1 hour since we fail the build after this amount of time anyway.